exporter/signalfxexporter/exporter.go

```diff
@@ -146,6 +143,29 @@ func newSignalFxExporter(
 	}, nil
 }
 
+func getHTTPClient(config *Config) (*http.Client, error) {
+	defaultMaxConns := 100
```

Comment:
I believe we want the defaults to be loaded in the default config (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/add40e0381a591accefacab8c14c39daa278b40b/exporter/signalfxexporter/factory.go#L54). The actual default values should probably be defined as const.

Reply:
As per the approach discussed, we will consider `maxIdleConns` value if provided, else take `maxConnections`’s value if provided and if not then we will assign a default value. So if we assign a default value in defaultConfig, and if the user has not provided the value of maxIdleConns then instead of checking the value of maxConnections it will directly take the default value of maxIdleConns from defaultconfig. So to check if the user has provided the value or not, we have not assigned the default value in defaultConfig and are assigning it in `getHTTPClient` function.

Yeah, I can define the default value as const but we still have to assign that const to a variable in this function as we have to assign an address to `maxIdleConns` as `maxIdleConns` is a pointer.